loki/transformations/single_column/vector.py

```diff
@@ -138,7 +138,12 @@ def extract_vector_sections(cls, section, horizontal):
                 subsec_body = cls.extract_vector_sections(separator.body, horizontal)
                 if subsec_body:
                     subsections += subsec_body
-                subsec_else = cls.extract_vector_sections(separator.else_body, horizontal)
+                # we need to prevent that the whole 'else_body' is wrapped in a section,
+                # as 'Conditional's rely on the fact that the first element of the 'else_body'
+                # (if 'has_elseif') is a Conditional itself
+                subsec_else = cls.extract_vector_sections(separator.else_body[0].body, horizontal)\
+                        if separator.else_body and isinstance(separator.else_body[0], ir.Conditional)\
+                        else cls.extract_vector_sections(separator.else_body, horizontal)
```

Comment:
Note that this is only required if the `Conditional` actually has `has_elseif == True`.
The following two version will create the same IR, with the subtle difference that the outer conditional has `has_elseif == True`
```
IF (A > 0) THEN
  A = A + 1
ELSE
  IF (A < 0) THEN
    A = A - 1
  ENDIF
ENDIF
```
vs.
```
IF (A > 0) THEN
  A = A + 1
ELSEIF (A < 0) THEN
  A = A - 1
ENDIF
```

For readability, I would also prefer to use proper if/else blocks rather than putting everything in a single line with line breaks. I would suggest rewriting this as
```suggestion
                if separator.has_elseif and separator.else_body:
                    subsec_else = cls.extract_vector_sections(separator.else_body[0].body, horizontal)
                else:
                    subsec_else = cls.extract_vector_sections(separator.else_body, horizontal)
```

Reply:
Yes! There is just a problem with OMNI as the parsing of Conditionals is a bit different to FP and OFP and `has_elseif` is always `False` for OMNI. Thus I could do what you suggest and disable OMNI or "fix" Conditional parsing/construction for OMNI.
What do you think?

Reply:
I don't think there is much you can do to "fix OMNI" in this case, as it automatically expands every `ELSEIF` into a nested `ELSE => IF` code structure in its frontend, even before we get the AST. So, unless we reconstruct the correct ELSEIF by inspecting the source string, there is no catching this. Usually, I'd say this leaves only special-casing the test, but since this quickly becomes cumbersome, I think we can just disable this test (with an explicit mention of OMNI's elseif-expansion).

Reply:
Agreed! The transformation's behaviour should be governed by the IR it receives, and not try to restore what OMNI has expanded previously. Importantly, OMNI does not change the meaning, simply the appearance. So, disabling the test for Omni (or hacking the IR by setting the flag before calling the transformation if you think that's better) seems sensible to me.